src/lib/cache/gdfs_cache.hpp

```diff
@@ -130,9 +157,9 @@ class GDFSCache : public AbstractCacheImpl<Key, Value> {
     return (*it->second).priority;
   }
 
-  ErasedIterator begin() { return ErasedIterator{std::make_unique<Iterator>(_map.begin())}; }
+  ErasedIterator unsafe_begin() { return ErasedIterator{std::make_unique<Iterator>(_map.begin())}; }
 
-  ErasedIterator end() { return ErasedIterator{std::make_unique<Iterator>(_map.end())}; }
+  ErasedIterator unsafe_end() { return ErasedIterator{std::make_unique<Iterator>(_map.end())}; }
 
   size_t frequency(const Key& key) {
     const auto it = _map.find(key);
```

Comment:
Why is this (as well as priority and potentially others) lock-free?

Reply:
I just copied the behavior of the old implementation, which guaranteed only the functions defined by the cache abstractions to be thread-safe. I can also make everything safe.

Reply:
Sounds better